### PR TITLE
Use  Awaitility  instead Thread.sleep in InactiveTopicDeleteTest.java

### PR DESCRIPTION
**Pull request by 315157973**
In many cases, we don’t need to sleep for that long. Using `Awaitility` can make the test faster

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by lhotari** [comment]
Good work!

**Comment by lhotari** [comment]
It looks like that in some cases, there could be a benefit of using `untilAsserted` and placing the test assertions in this block. The pattern is useful when there are multiple conditions. It also reduces the need to have duplicate code for the assertions (compared to the case of using `until`).

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java [hunk 1]
```diff
@@ -82,12 +84,14 @@ public void testDeleteWhenNoSubscriptions() throws Exception {
         consumer.close();
         producer.close();
 
-        Thread.sleep(2000);
+        Awaitility.await().atMost(2, TimeUnit.SECONDS).until(() -> admin.topics().getList("prop/ns-abc")
+                .contains(topic));
         Assert.assertTrue(admin.topics().getList("prop/ns-abc")
             .contains(topic));
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:90]: It can simplify by untiAssert(), Please check all.

Comment (315157973) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:90]: Thanks lhotari for the reminder

**Review by jiazhai — APPROVED** [review]

**Comment by jiazhai** [comment]
Thanks @315157973 for the help.

**Comment by 315157973** [comment]
> Thanks @315157973 for the help.

My pleasure :smile:

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java [hunk 2]
```diff
@@ -538,13 +523,8 @@ public void testDeleteWhenNoSubscriptionsWithTopicLevelPolicies() throws Excepti
...
-            if (admin.topics().getInactiveTopicPolicies(topic2) != null) {
-                break;
-            }
-            Thread.sleep(100);
-        }
-
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).until(()
+                -> admin.topics().getInactiveTopicPolicies(topic2) != null);
         // topic should still exist
         Thread.sleep(2000);
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:529]: This one should also can instead by Awaitility

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**